### PR TITLE
Add certificate service authentication

### PR DESCRIPTION
**Pull request by sbabev**

### CODE REVIEW

File: src/authentication.ts [hunk 1]
```diff
@@ -169,3 +180,37 @@ export class OAuthAuthentication implements Authentication {
         });
     }
 }
+
+/**
+ * Basic Authentication class. Retrieves the basic authorization header value.
```
Comment (yusmenrashidov) [inline src/authentication.ts:185]: Certificate Authentication?

File: src/utils/region.ts [hunk 2]
```diff
@@ -21,6 +21,45 @@ const ANS_CA10_INSTANCE = 'https://clm-sl-ans-live-ans-service-api.cfapps.ca10.h
...
+const ANS_AP12_MESH_INSTANCE =
+    'https://clm-sl-ans-live-ans-service-api.mesh.cf.ap12.hana.ondemand.com';
+const ANS_AP20_MESH_INSTANCE =
+    'https://clm-sl-ans-live-ans-service-api.mesh.cf.ap20.hana.ondemand.com';
+const ANS_AP21_MESH_INSTANCE =
+    'https://clm-sl-ans-live-ans-service-api.mesh.cf.ap21.hana.ondemand.com';
+const ANS_BR10_MESH_INSTANCE =
+    'https://clm-sl-ans-live-ans-service-api.mesh.cf.br10.hana.ondemand.com';
+const ANS_EU10_MESH_INSTANCE =
+    'https://clm-sl-ans-canary-ans-service-api.mesh.cf.sap.hana.ondemand.com';
```
Comment (yusmenrashidov) [inline src/utils/region.ts:37]: `'https://clm-sl-ans-canary-ans-service-api.mesh.cf.eu10.hana.ondemand.com';`

**Review by yusmenrashidov — DISMISSED** [review]

File: src/authentication.ts [hunk 3]
```diff
@@ -169,3 +180,37 @@ export class OAuthAuthentication implements Authentication {
         });
     }
 }
+
+/**
+ * Certificate service Authentication class. Provides the certificate and the private key for the building of the keystore.
+ */
+export class CertificateServiceAuthentication {
```
Comment (mkunev) [inline src/authentication.ts:187]: Rename to CertificateAuthentication and fix all documentation (including comments) related to it

File: src/authentication.ts [hunk 4]
```diff
@@ -27,6 +27,17 @@ export interface OAuthConfig {
     oAuthTokenUrl: string;
 }
 
+export interface CertificateServiceConfig {
```
Comment (mkunev) [inline src/authentication.ts:30]: rename to CertificateConfig

File: docs/authentication/certificate-service-authentication.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+{: .no_toc .text-delta }
+
+1. TOC
+   {:toc}
+
+---
+
+## Description
+
+Certificate service Authentication is a class in the context of Alert Notification service node client. It helps you to authenticate directly against Alert Notification service.
```
Comment (mkunev) [inline docs/authentication/certificate-service-authentication.md:22]: CertificateAuthentication is a class in the context of Alert Notification service node client. It is used to trigger the mTLS based authentication, you just need to provide the certificate and private key generated from your Alert Notification service instance (e.g. from a Service Key or Service Binding). 

Also add a link to help.sap.com for further reading

File: docs/_data/certificate-service-authentication.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+  "propertyFields": [
+    {
+      "name": "certificate",
+      "type": "string",
+      "description": "A tecnical client's certificate"
```
Comment (mkunev) [inline docs/_data/certificate-service-authentication.json:6]: x509 certificate chain generated by Alert Notification service

File: docs/_data/certificate-service-authentication.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    {
+      "name": "privateKey",
+      "type": "string",
+      "description": "A technical client's privateKey"
```
Comment (mkunev) [inline docs/_data/certificate-service-authentication.json:11]: Private key related to the generated certificate

File: docs/authentication/certificate-service-authentication.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
+---
+layout: default
+title: Certificate service Authentication
```
Comment (mkunev) [inline docs/authentication/certificate-service-authentication.md:3]: mTLS Authentication/Certificate Authentication

File: docs/authentication/certificate-service-authentication.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+parent: Authentication
+nav_order: 2
+permalink: /authentication/certificate-service
```
Comment (mkunev) [inline docs/authentication/certificate-service-authentication.md:6]: /authentication/mtls-authentication or /authentication/certificate-authnetication depends on the title you choose

File: docs/authentication/certificate-service-authentication.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (mkunev) [inline docs/authentication/certificate-service-authentication.md:1]: rename file to match the title of the page

File: docs/authentication/certificate-service-authentication.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+| Name | Returns | Description |
+|:---:|:--:|:---------:| {% for method in site.data.certificate-service-authentication.methods %}
+| {{method.name}} | {{method.returnValue}}|{{method.description}} | {% endfor %}
+
+## @Example
+
+```js
+import { CertificateServiceAuthentication } from '@sap_oss/alert-notification-client';
+
+const certificateServiceAuthentication = new CertificateServiceAuthentication({
```
Comment (mkunev) [inline docs/authentication/certificate-service-authentication.md:45]: variable should be changed accordingly when class is renamed

File: docs/authentication/authentication-overview.md [hunk 12]
```diff
@@ -13,7 +13,7 @@ Alert Notification service supports two types of authentication:
 
 * _Basic_
 * _OAuth_
-
+* _Certificate service_
```
Comment (mkunev) [inline docs/authentication/authentication-overview.md:16]:
```suggestion
* _mTLS_
```

File: src/authentication.ts [hunk 13]
```diff
@@ -169,3 +180,37 @@ export class OAuthAuthentication implements Authentication {
...
+     *
+     * @param {Credentials} creds - object which contains the certificate and private key. Certificate and privateKey
+     * must be provided else an Error will be thrown.
+     */
+    constructor(creds: CertificateConfig) {
+        if (!creds) {
+            throw new Error('Credentials must not be null or undefined');
+        }
+
+        if (!creds.certificate && !creds.privateKey) {
```
Comment (mkunev) [inline src/authentication.ts:201]:
```suggestion
        if (!creds || !creds.certificate || !creds.privateKey) {
```

File: src/authentication.ts [hunk 14]
```diff
@@ -169,3 +180,37 @@ export class OAuthAuthentication implements Authentication {
...
+            throw new Error('Certificate and privateKey must be provided');
```
Comment (mkunev) [inline src/authentication.ts:202]:
```suggestion
            throw new Error('Both certificate and privateKey must be provided when using CertificateAuthentication');
```

**Review by mkunev — APPROVED** [review]

**Review by yusmenrashidov — APPROVED** [review]

**Pull request merged**